src/sync.c

```diff
@@ -33,44 +33,35 @@ static void dump_ref(uint8_t *ref_buf)
     // log_debug("REF %08X", value);
 }
 
-static void calc_phase(float complex *buf, unsigned int ref, float *out_phase, float *out_slope)
+static void adjust_ref(float complex *buf, float *phases, unsigned int ref)
 {
+    // sync bits (after DBPSK)
+    static const signed char sync[] = {
+        -1, 1, -1, -1, -1, 1, 1
+    };
     float phase, slope;
     float complex sum;
 
     sum = 0;
-    for (int r = 0; r < BLKSZ; r++)
-        sum += buf[ref * BLKSZ + r] * buf[ref * BLKSZ + r];
-    phase = cargf(sum) * 0.5;
```

Comment:
This method of calculating the phase only works when the slope is very small. Error in the sampling frequency increases the slope, so I had to change it to take the slope into account.